File: 백준/Silver/1926. 그림/그림.py
import sys 
from collections import deque
input = sys.stdin.readline 
sys.setrecursionlimit(10**6)

def bfs(x, y):
    graph[x][y] = 0
    dx = [1, -1, 0, 0]
    dy = [0, 0, 1, -1]
    width = 1
    q = deque()
    q.append([x, y])
    while q:
        x, y = q.popleft()
        for i in range(4):
            nx = dx[i] + x
            ny = dy[i] + y
            if 0 <= nx < n and 0 <= ny < m and graph[nx][ny] == 1:
                q.append([nx, ny])
                graph[nx][ny] = 0
                width += 1
    return width


n, m = map(int, input().split())
graph = [list(map(int, input().split())) for _ in range(n)]
cnt = 0
res = 0
for i in range(n):
    for j in range(m):
        if graph[i][j] == 1:
            cnt += 1
            res = max(bfs(i, j), res)
print(cnt)
print(res)

File: 백준/Silver/1926. 그림/그림.go
package main

import (
	"bufio"
	"container/list"
	"fmt"
	"os"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

type Queue struct {
	v *list.List
}

func NewQueue() *Queue {
	q := Queue{list.New()}
	return &q
}
func (q *Queue) Push(n []int) {
	q.v.PushBack(n)
}

func (q *Queue) Pop() []int {
	f := q.v.Front()
	if f == nil {
		return nil
	}
	return q.v.Remove(f).([]int)
}

func (q *Queue) Empty() bool {
	if q.v.Len() == 0 {
		return true
	}
	return false
}

var (
	graph [][]int
	dx    = []int{1, -1, 0, 0}
	dy    = []int{0, 0, 1, -1}
	n, m  int
)

func bfs(x, y int) int {

	graph[x][y] = 0
	var width = 1

	q := NewQueue()
	q.Push([]int{x, y})
	for !q.Empty() {
		num := q.Pop()
		for i := 0; i < 4; i++ {
			nx := dx[i] + num[0]
			ny := dy[i] + num[1]
			if 0 <= nx && nx < n && 0 <= ny && ny < m && graph[nx][ny] == 1 {
				q.Push([]int{nx, ny})
				graph[nx][ny] = 0
				width += 1
			}
		}
	}
	return width
}

func main() {
	defer writer.Flush()
	fmt.Fscanf(reader, "%d %d\n", &n, &m)
	graph = make([][]int, n)
	for i := 0; i < n; i++ {
		graph[i] = make([]int, m)
		for j := 0; j < m; j++ {
			fmt.Fscanf(reader, "%d ", &graph[i][j])
		}
	}
	cnt := 0
	res := 0
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			if graph[i][j] == 1 {
				cnt += 1
				res = max(bfs(i, j), res)
			}
		}
	}
	fmt.Fprintf(writer, "%d\n%d", cnt, res)
}
